andremun/pyInstanceSpace
# Architecture Planning v2 (Previously Options Overhaul) (#80)

* propose options method

* improve docs for options

* make Metadata consistant

* make Metadata consistant

* Data can be changed

* add function to instantiate Options with mostly default values

* fix poe errors

* fix rename of options

* add data to outs

* remove the output of future stages if a previous stage is ran

* add raises when required options aren't present in cloister

* create a from_options function for prelim options

* unify return values of stages

* fix mypy and pytest clash, 1 error in tests

* fix mypy error

* make Data immutable

* make stage state order of arguments consistant with stage output

* remove file handling from this pr

* remove file handling from this pr

* add defaults from example.m

* name default options module something better

* fix poe errors

* remove logging if options where none, as they can't be none

* clean up with poe check

---------
